File: src/automatica.core/Automatica.WebNew/src/app/base/model/web-api-exception.ts
import { BaseModel, Model, JsonFieldInfo, JsonProperty, JsonPropertyName } from "./base-model";

export enum ExceptionSeverity {
    Info,
    Warning,
    Error,
    Dead
}

@Model()
export class WebApiException extends BaseModel {
    private _errorText: string;
    private _severity: ExceptionSeverity;

    @JsonPropertyName("severity")
    public get Severity(): ExceptionSeverity {
        return this._severity;
    }
    public set Severity(v: ExceptionSeverity) {
        this._severity = v;
    }


    @JsonPropertyName("errorText")
    public get ErrorText(): string {
        return this._errorText;
    }
    public set ErrorText(v: string) {
        this._errorText = v;
    }



    public typeInfo() {
        return "WebApiException";
    }

    protected getJsonProperty(): Map<string, JsonFieldInfo> {
        return void 0;
    }
}

File: src/automatica.core/Automatica.Core.Base/Exceptions/WebApiException.cs
using System;
using System.Dynamic;

namespace Automatica.Core.Base.Exceptions
{

    public enum ExceptionSeverity
    {
        Info,
        Warning,
        Error,
        Dead
    }

    /// <summary>
    /// Provides a serializable exception for the UI
    /// </summary>
    [Serializable]
    public class WebApiException : Exception
    {
        public WebApiException(string errorText, ExceptionSeverity severity)
        {
            ErrorText = errorText;
            Severity = severity;
        }
        public string ErrorText { get; }
        public ExceptionSeverity Severity { get; }


        public object ToJson()
        {
            dynamic obj = new ExpandoObject();
            obj.errorText = ErrorText;
            obj.severity = Severity;

            obj.TypeInfo = "WebApiException";
            return obj;
        }
    }
}
